polaris-core/src/main/java/org/apache/polaris/core/persistence/pagination/PageTokenUtil.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.polaris.core.persistence.pagination;
+
+import static com.google.api.client.util.Preconditions.checkArgument;
+
+import jakarta.annotation.Nullable;
+import java.util.Optional;
+
+public final class PageTokenUtil {
+
+  private PageTokenUtil() {}
+
+  public static PageRequest decodePageRequest(
+      @Nullable String requestedPageToken, @Nullable Integer requestedPageSize) {
+    int pageSize;
+    String token = null;
+    if (requestedPageToken != null) {
+      int idx = requestedPageToken.indexOf(':');
```

Comment:
I know it's not a new thing to this PR, but I don't understand why the size of the _previous_ page matters at all.
AFAIK the page-size is included in every request. Feels a bit weird to have an "outer" and "inner" meaning of an otherwise opaque token.

Reply:
> AFAIK the page-size is included in every request

That's not correct

Reply:
I'll keep the propagation of the previous page size into the next request via the page token. 

I think doing that is reasonable to maintain better alignment with the REST API spec.